esp-hal-common/src/soc/esp32s2/mod.rs

```diff
@@ -26,3 +30,57 @@ pub(crate) mod constants {
     pub const SOC_DRAM_LOW: u32 = 0x3FFB_0000;
     pub const SOC_DRAM_HIGH: u32 = 0x4000_0000;
 }
+
+/// Function initializes ESP32 specific memories (RTC slow and fast) and
+/// then calls original Reset function
+///
+/// ENTRY point is defined in memory.x
+/// *Note: the pre_init function is called in the original reset handler
+/// after the initializations done in this function*
+#[cfg(feature = "rt")]
```

Comment:
there is no such feature in esp-hal-common so we don't get ESP32Reset function - makes the code crash immediately